cmd/kots/cli/discover.go

```diff
@@ -0,0 +1,201 @@
+package cli
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func discover() (string, string, string) {
+	var dist string
+	var version string
+
+	var lsbDist string
+	var distVersion string
+	var distVersionMajor string
+	var majorVersion string
+
+	// Find the Linux Standard Distribution
+	if _, err := os.Stat("/etc/centos-release"); err == nil {
+		// this is for centos
+		distCmd := fmt.Sprintf("%s", "cat /etc/os-centos-release | cut -d\" \" -f1")
+		dist, _ = getBashData("/etc/os-centos-release", distCmd)
+
+		// Get Version
+		versionCmd := fmt.Sprintf("%s", "cat /etc/centos-release | sed 's/Linux //' | cut -d\" \" -f3 | cut -d \".\" -f1-2")
+		version, _ = getBashData("/etc/os-centos-release", versionCmd)
+	} else if _, err := os.Stat("/etc/redhat-release"); err == nil {
+		// this is for rhel6
+		dist = "rhel"
+
+		// Get Version
+		majorVersionCmd := fmt.Sprintf("%s", "cat /etc/redhat-release | cut -d\" \" -f7 | cut -d \".\" -f1")
+		majorVersion, _ = getBashData("/etc/redhat-release", majorVersionCmd)
+		version = majorVersion
+	} else if _, err := os.Stat("/etc/os-release"); err == nil {
+		// ubuntu
+		// Get dist
+		cmd := exec.Command("/bin/bash", "-c", "source /etc/os-release ; echo $ID;")
+		output, err := cmd.CombinedOutput()
+		if err == nil {
+			dist = string(output)
+		}
+		// Get Version
+		cmd = exec.Command("/bin/bash", "-c", "source /etc/os-release ; echo $VERSION_ID;")
+		output, err = cmd.CombinedOutput()
+		if err == nil {
+			version = string(output)
+		}
+	}
+
+	// Remove newlines if any
+	dist = strings.TrimSuffix(dist, "\n")
+	version = strings.TrimSuffix(version, "\n")
+
+	// convert into lower
+	dist = strings.ToLower(dist)
+	if isValidDistribution(dist) == true {
+		switch dist {
+		case "ubuntu":
+			major := strings.Split(version, ".")
+			if v, _ := strconv.Atoi(major[0]); v > 12 {
```

Comment:
Are these really the supported versions for Kots? They look like the supported versions of our old platform product.